SparseGrids/tsgRuleLocalPolynomial.hpp

```diff
@@ -261,6 +265,86 @@ class templRuleLocalPolynomial : public BaseRuleLocalPolynomial{
             }
         }
     }
+
+    double diffRaw(int point, double x) const override{
+        if (isZeroOrder) {
+            return 0.0;
+        } else {
+            if (rule == rule_localp or rule == rule_semilocalp) {
+                if (point == 0) return 0.0;
+                if (rule == rule_semilocalp) {
+                    if (point == 1) return x - 0.5;
+                    if (point == 2) return x + 0.5;
+                }
+            }
+            double xn = scaleX(point, x);
+            double an = scaleDiffX(point);
+            if (rule != rule_semilocalp and max_order == 1) {
+                if (x == 1.0) {
+                    if (rule == rule_localp0 and point == 0)
+                        return  -1.0;
+                    else if (rule == rule_localpb and point == 1)
+                        return 0.5;
+                    else if (rule == rule_localpb and point == 2)
+                        return -1.0;
+                    else if (point == 2)
+                        return an;
+                }
+                return (xn >= 0 ? -1.0 : 1.0) * an;
+            }
+            if (max_order == 2) return an * diffPWQuadratic(point, xn);
+            if (max_order == 3) return an * diffPWCubic(point, xn);
+            return an * diffPWPower(point, xn);
+        }
+    }
+
+    double diffSupport(int point, double x, bool &isSupported) const override{
+        if (isZeroOrder) {
+            isSupported = false;
+            return 0.0;
+        } else {
+            // We still need (isSupported == true) at point = 0 so that TasGrid::GridLocalPolynomial::walkTree<3>()
+            // walks through the children of this point.
+           isSupported = true;
+            if (rule == rule_localp or rule == rule_semilocalp) {
+                if (point == 0) {
+                    return 0.0;
+                }
+                if (rule == rule_semilocalp) {
+                    if (point == 1) return x - 0.5;
+                    if (point == 2) return x + 0.5;
+                }
+            }
+            // Note that the support will be of the form [a, b), except for x = +1.0, i.e., the rightmost node in the domain. This
+            // is to avoid double-counting derivatives at points of discontinuity.
+            double xn = scaleX(point, x);
+            bool rightmost_node = (x == 1.0 and xn == 1.0);
```

Comment:
If `x == 1.0` then how can it be that `xn != 1.0`? The if-condition seems redundant.

Reply:
This happens in many cases. If `x == 1.0` and the support associated with `point` does not contain `1.0`, then `xn > 1.0`. So the bool `rightmost_node` is only true if `x` is the rightmost point in the domain associated with `point`.

Reply:
So your check is whether you're using the right-most basis function combined with the right-most domain point, one relates to `x` and the other to `point`. The name is a bit confusing, what is `rightmost` and how that relates to the support.

Why not combine with the next line:
```
isSupported = (-1.0 <= xn and xn < 1.0) or (xn == 1.0 and x == 1.0);
```
Now we can see that we consider the interval `xn \in [-1.0, 1.0)` but make an exception for the edge case when `x==1.0` we use `xn \in [-1.0, 1.0]`

Reply:
Sure. I've changed it to 
```
isSupported = (-1.0 <= xn and xn < 1.0) or (xn == 1.0 and x == 1.0);
```
in the latest push.